bulk-ingester/src/test/java/org/alfresco/hxi_connector/bulk_ingester/processor/mapper/AlfrescoPropertyMapperTest.java

```diff
@@ -0,0 +1,489 @@
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.NODEREF;
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.NULL;
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.PATH;
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.SERIALIZABLE;
+import static org.alfresco.elasticsearch.db.connector.model.PropertyValueType.STRING;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.util.SerializationUtils;
+
+import org.alfresco.elasticsearch.db.connector.model.AlfrescoNode;
+import org.alfresco.elasticsearch.db.connector.model.ContentMetadata;
+import org.alfresco.elasticsearch.db.connector.model.NodeProperty;
+import org.alfresco.elasticsearch.db.connector.model.PropertyKey;
+import org.alfresco.elasticsearch.db.connector.model.PropertyValue;
+import org.alfresco.elasticsearch.db.connector.model.PropertyValueType;
+
+@SuppressWarnings({"PMD.TooManyMethods", "PMD.LinguisticNaming"})
+class AlfrescoPropertyMapperTest
+{
+    @Test
+    void shouldMapJustPropertiesWithGivenName()
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String description = "description";
+        String descriptionText = "The purpose of document is...";
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty("title"),
+                        createNodeProperty(description, stringValue(descriptionText))));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, description).performMapping();
+
+        // then
+        assertTrue(property.isPresent());
+        assertEquals(description, property.get().getKey());
+        assertEquals(descriptionText, property.get().getValue());
+    }
+
+    @Test
+    void shouldProcessPropertiesWithSameNameTogether()
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String description = "description";
+        String usDescription = "French fries recipe";
+        String ukDescription = "Chips recipe";
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty("title"),
+                        createNodeProperty(description, "en_US_", stringValue(usDescription)),
+                        createNodeProperty(description, "en_UK_", stringValue(ukDescription))));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, description).performMapping();
+
+        // then
+        assertTrue(property.isPresent());
+        assertEquals(description, property.get().getKey());
+        assertThat(List.of(usDescription, ukDescription)).hasSameElementsAs((Iterable<String>) property.get().getValue());
```

Comment:
I think the two arguments are backwards - `assertThat(actual).hasSameElementsAs(actual)`

Reply:
I think you meant `assertThat(actual).hasSameElementsAs(expected)`

Reply:
that's right, fixed